components/editor/CodeBlock.tsx

```diff
@@ -0,0 +1,342 @@
+    fontStyle: 'italic'
+  },
+  'hljs-formula': {
+    color: '#7edcda'
+  },
+  'hljs-strong': {
+    fontWeight: 'bold'
+  },
+  'hljs-link:hover': {
+    textDecoration: 'underline'
+  },
+  'hljs-quote': {
+    color: '#797f8c'
+  },
+  'hljs-comment': {
+    color: '#797f8c'
+  },
+  'hljs-doctag': {
+    color: '#7edcda'
+  },
+  'hljs-$ref': {
+    color: 'yellow'
+  },
+  'hljs-meta': {
+    color: '#5E81AC'
+  },
+  'hljs-meta-keyword': {
+    color: '#5E81AC'
+  },
+  'hljs-meta-string': {
+    color: '#c0e2a3'
+  },
+  'hljs-attr': {
+    color: '#7edcda'
+  },
+  'hljs-attribute': {
+    color: '#d6deeb'
+  },
+  'hljs-builtin-name': {
+    color: '#64a0dc'
+  },
+  'hljs-name': {
+    color: '#64a0dc'
+  },
+  'hljs-section': {
+    color: '#74e287'
+  },
+  'hljs-tag': {
+    color: '#64a0dc'
+  },
+  'hljs-variable': {
+    color: '#d6deeb'
+  },
+  'hljs-template-variable': {
+    color: '#d6deeb'
+  },
+  'hljs-template-tag': {
+    color: '#5E81AC'
+  },
+  'yaml .hljs-meta': {
+    color: '#D08770'
+  }
+};
+
+/**
+ * @description This component displays code snippets with syntax highlighting.
+ *
+ * @param {CodeBlockProps} props - The component props.
+ * @param {string} props.children - The code snippet to be displayed.
+ * @param {Array<{ code: string; title?: string; language?: string }>} props.codeBlocks - An array of code blocks
+ *  with optional titles and languages.
+ * @param {string} props.className - Additional CSS class for styling the CodeBlock component.
+ * @param {string} props.highlightClassName - Additional CSS class for styling the code highlighting area.
+ * @param {number[]} props.highlightedLines - An array of line numbers to be highlighted.
+ * @param {string} props.language - The programming language for syntax highlighting (default is 'yaml').
+ * @param {boolean} props.hasWindow - Indicates whether window controls should be displayed.
+ * @param {boolean} props.showCopy - Indicates whether the copy-to-clipboard button should be displayed.
+ * @param {boolean} props.showCaption - Indicates whether the caption should be displayed.
+ * @param {string} props.caption - The caption text to be displayed.
+ * @param {boolean} props.showLineNumbers - Indicates whether line numbers should be displayed.
+ * @param {number} props.startingLineNumber - The starting line number for line numbering.
+ * @param {string} props.textSizeClassName - Additional CSS class for controlling the text size.
+ * @param {string} props.title - The title of the code block (default is the specified language).
+ */
+export default function CodeBlock({
+  children,
+  codeBlocks,
+  className = '',
+  highlightClassName = '',
+  highlightedLines,
+  language = 'yaml',
+  hasWindow = false,
+  showCopy = true,
+  showCaption = true,
+  caption = '',
+  showLineNumbers = true,
+  startingLineNumber = 1,
+  textSizeClassName = 'text-xs',
+  // eslint-disable-next-line no-unused-vars, unused-imports/no-unused-vars
+  title = language
```

Comment:
Since, we are not using title, we can remove it, right?

Reply:
Done